packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -378,6 +371,617 @@ function setProp(
+      break;
+    case 'vertOriginY':
+      setValueForAttribute(domElement, 'vert-origin-y', value);
+      break;
+    case 'wordSpacing':
+      setValueForAttribute(domElement, 'word-spacing', value);
+      break;
+    case 'writingMode':
+      setValueForAttribute(domElement, 'writing-mode', value);
+      break;
+    case 'xmlnsXlink':
+      setValueForAttribute(domElement, 'xmlns:xlink', value);
+      break;
+    case 'xHeight':
+      setValueForAttribute(domElement, 'x-height', value);
+      break;
+    case 'xlinkActuate':
+      setValueForNamespacedAttribute(
+        domElement,
+        xlinkNamespace,
+        'xlink:actuate',
+        value,
+      );
+      break;
+    case 'xlinkArcrole':
+      setValueForNamespacedAttribute(
+        domElement,
+        xlinkNamespace,
+        'xlink:arcrole',
+        value,
+      );
+      break;
+    case 'xlinkRole':
+      setValueForNamespacedAttribute(
+        domElement,
+        xlinkNamespace,
+        'xlink:role',
+        value,
+      );
+      break;
+    case 'xlinkShow':
+      setValueForNamespacedAttribute(
+        domElement,
+        xlinkNamespace,
+        'xlink:show',
+        value,
+      );
+      break;
+    case 'xlinkTitle':
+      setValueForNamespacedAttribute(
+        domElement,
+        xlinkNamespace,
+        'xlink:title',
+        value,
+      );
+      break;
+    case 'xlinkType':
+      setValueForNamespacedAttribute(
+        domElement,
+        xlinkNamespace,
+        'xlink:type',
+        value,
+      );
+      break;
+    case 'xmlBase':
+      setValueForNamespacedAttribute(
+        domElement,
+        xmlNamespace,
+        'xml:base',
+        value,
+      );
+      break;
+    case 'xmlLang':
+      setValueForNamespacedAttribute(
+        domElement,
+        xmlNamespace,
+        'xml:lang',
+        value,
+      );
+      break;
+    case 'xmlSpace':
+      setValueForNamespacedAttribute(
+        domElement,
+        xmlNamespace,
+        'xml:space',
+        value,
+      );
+      break;
+    // Properties that should not be allowed on custom elements.
+    case 'innerText':
+    case 'textContent':
+      if (enableCustomElementPropertySupport) {
+        break;
+      }
+    // eslint-disable-next-line no-fallthrough
+    default: {
+      if (
+        key.length > 2 &&
+        (key[0] === 'o' || key[0] === 'O') &&
+        (key[1] === 'n' || key[1] === 'N')
+      ) {
+        if (
+          __DEV__ &&
+          registrationNameDependencies.hasOwnProperty(key) &&
+          value != null &&
+          typeof value !== 'function'
+        ) {
+          warnForInvalidEventListener(key, value);
+        }
+      } else {
+        setValueForAttribute(domElement, key, value);
+      }
+    }
+  }
+}
+
+function setPropOnCustomElement(
+  domElement: Element,
+  tag: string,
+  key: string,
+  value: mixed,
+  props: any,
+): void {
+  switch (key) {
+    case 'style': {
```

Comment:
This is a replica of the beginning of setProps. We can probably unify some of this.